assets/src/design-system/components/slider/index.js

```diff
@@ -0,0 +1,255 @@
+// This is used to align the tooltip by, as this is the base position at the 0 mark.
+const FakeThumb = styled.div`
+  width: 0;
+  height: 0;
+  position: absolute;
+  left: ${({ thumbSize }) => thumbSize / 2}px;
+  top: ${({ thumbSize }) => -thumbSize / 2 + 12}px;
+`;
+
+/**
+ * A styled range input component.
+ *
+ * This component must be initialized with two step values - one value (`majorStep`)
+ * is the coarse value, that simply pressing arrow-left and arrow-right will move
+ * between (e.g. 1) and the other (`minorStep`) is the more fine-grained value (e.g. 0.1)
+ * which can be used by pressing shift+arrow.
+ *
+ * When using the mouse, only `minorStep` is considered and this is the resolution the
+ * range has.
+ *
+ * @param {Object} props Properties
+ * @param {number} props.value Current value
+ * @param {Function} props.handleChange Callback when updated
+ * @param {number} props.majorStep Major step as described
+ * @param {number} props.minorStep Minor step as described
+ * @param {number} props.min Minimum value
+ * @param {number} props.max Maximum value
+ * @param {number} props.thumbSize Thumb diameter in pixels if different from standard
+ * @param {string} props.suffix Optional value suffix to be displayed in tooltip
+ * @return {Node} Range input component
+ */
+function Slider({
+  minorStep,
+  majorStep,
+  handleChange,
+  value,
+  min = 0,
+  max = 500,
+  thumbSize = DEFAULT_SIZE,
+  suffix = '',
+  ...rest
+}) {
+  const ref = useRef();
+  const fakeThumbRef = useRef();
+  const widthTracker = useRef(null);
+  const update = useCallback(
+    (direction, isMajor) => {
+      const diff = direction * (isMajor ? majorStep : minorStep);
+      let val = value + diff;
+      val = typeof min === 'number' ? Math.max(min, val) : val;
+      val = typeof max === 'number' ? Math.min(max, val) : val;
+      handleChange(val);
+    },
+    [minorStep, majorStep, handleChange, min, max, value]
+  );
+
+  const percentageVal = ((value - min) / (max - min)) * 100;
+
+  useEffect(() => {
+    if (ref.current && !widthTracker.current) {
+      const { width } = ref.current.getBoundingClientRect();
+      widthTracker.current = width;
+    }
+  }, [ref]);
+
+  useKeyDownEffect(ref, ['left', 'down'], () => update(-1, true), [update]);
+  useKeyDownEffect(ref, ['right', 'up'], () => update(1, true), [update]);
+  useKeyDownEffect(ref, ['shift+left', 'shift+down'], () => update(-1, false), [
+    update,
+  ]);
+  useKeyDownEffect(ref, ['shift+right', 'shift+up'], () => update(1, false), [
+    update,
+  ]);
+
+  const printValue = `${value}${suffix}`;
+
+  return (
+    <Tooltip
+      title={printValue}
+      placement={PLACEMENT.TOP}
+      forceAnchorRef={fakeThumbRef}
+      tooltipProps={{
+        style: {
+          transform: `translate(${
+            ((widthTracker.current - thumbSize) * percentageVal) / 100
+          }px, 0)`,
+        },
+      }}
```

Comment:
This is far from pretty, but not sure how to clean it up. I prefer setting style on the tooltip rather than some specific `offsetX` property, as this is a lot more flexible and less verbose, but this is also a pretty nasty expression.